resolwe/flow/executors/prepare.py

```diff
@@ -0,0 +1,49 @@
+"""Base class for the manager-internal preparation bits of the executor."""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+import os
+
+from django.apps import apps
+from django.conf import settings
+from django.forms.models import model_to_dict
+
+from resolwe.flow.models import Data
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+CWD = os.getcwd()
+
+
+class BaseFlowExecutorPreparer(object):
+    """Represents the preparation functionality of the workflow executor."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize attributes."""
+
+    def make_settings(self, data_id):
+        """Add any additional required settings into what the manager will serialize."""
+        return {
+            'USE_TZ': settings.USE_TZ,
+            'FLOW_EXECUTOR_TOOLS_PATHS': self.get_tools(),
+            'Data_get': model_to_dict(Data.objects.get(
+                pk=data_id
+            )),
+            'Process_get': model_to_dict(Data.objects.get(
+                pk=data_id
+            ).process),
+        }
+
+    def get_tools(self):
+        """Get tools paths."""
```

Comment:
As discussed, this logic should be moved to a manage.py command which collects tools and copies them to the predefined location, like `collectstatic` does with static files. Then this function can just return path to that dir.